api-scanner/src/main/java/net/corda/plugins/ScanApi.java

```diff
@@ -400,3 +420,17 @@ private static URL toURL(File file) throws MalformedURLException {
         return urls.toArray(new URL[urls.size()]);
     }
 }
+
+class Names {
+    List<String> visible;
+    @SuppressWarnings("WeakerAccess") List<String> hidden;
+
+    Names(List<String> visible, List<String> hidden) {
+        this.visible = unmodifiable(visible);
+        this.hidden = unmodifiable(hidden);
+    }
+
+    private static <T> List<T> unmodifiable(List<T> list) {
+        return list.isEmpty() ? emptyList() : unmodifiableList(new ArrayList<>(list));
```

Comment:
Abit excessive, no?

Reply:
`emptyList()` is already unmodifiable, so wrapping it with `unmodifiableList()` might cause me to wake up screaming in the middle of the night.

Reply:
I mean `unmodifiableList(list)` is sufficient

Reply:
I'm not sure about that, because `unmodifiableList(list)` could still be modified if someone had a reference to the original `list`.